File: slm-engine/src/exceptions/app_exception.py
from enums.response_enum import ResponseEnum

class AppException(Exception):
    def __init__(self, message_or_enum, code: int = None):
        if isinstance(message_or_enum, ResponseEnum):
            self.message = message_or_enum.message
            self.code = message_or_enum.code
        else:
            self.message = message_or_enum
            self.code = code if code is not None else ResponseEnum.INTERNAL_SERVER_ERROR.code

        super().__init__(self.message) # Pass both message and code

File: slm-embed/src/main/java/com/slm/slmembed/exception/AppException.java
package com.slm.slmembed.exception;

import com.slm.slmembed.enums.ResponseEnum;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class AppException extends RuntimeException {

    private int code;

    public AppException(int statusCode, String message) {
        super(message);
        this.code = statusCode;
    }

    public AppException(ResponseEnum responseEnum) {
        super(responseEnum.getMessage());
        this.code = responseEnum.getCode();
    }

    public AppException(ResponseEnum responseEnum, String message) {
        super(message);
        this.code = responseEnum.getCode();
    }
}